source/server/listener_manager_impl.cc

```diff
@@ -166,6 +166,11 @@ Network::SocketSharedPtr ProdListenerComponentFactory::createListenSocket(
                                  ? Network::Utility::TCP_SCHEME
                                  : Network::Utility::UDP_SCHEME;
   const std::string addr = absl::StrCat(scheme, address->asString());
+
+  if (!bind_to_port && socket_type == Network::Address::SocketType::Stream) {
```

Comment:
Is this needed? Won't this just return -1 below once you check for not binding in the parent?

Reply:
Here is the need to determine whether the bind, if there is no bind socket to the parent process to obtain the socket, may get the already bound socket, in addition, there is no way to handle the traffic without the bind socket, you can not go to the parent process to obtain the socket, directly Create it.